File: 13-RomanToInteger/13-RomanToInteger.py
# Last updated: 9/14/2025, 9:19:13 AM
class Solution:
    def romanToInt(self, s: str) -> int:
        romans={'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        last=len(s)-1
        tot=romans.get(s[-1])
        last_value=tot
        for i in reversed(range(len(s)-1)):
            current_value=romans.get(s[i])
            if current_value<last_value:
                tot-=current_value
            else:
                tot+=current_value
            last_value=current_value
        return tot

File: 13-RomanToInteger/13-RomanToInteger.java
// Last updated: 9/14/2025, 9:19:09 AM
class Solution {
    public int romanToInt(String s) {
        HashMap<Character,Integer> map = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);
        char []arr=s.toCharArray();
        int tot=map.get(arr[arr.length-1]);
        int prev=tot;
        for(int i=arr.length-2;i>=0;i--){
            int curr=map.get(arr[i]);
            if(curr>=prev){
                tot+=curr;
            }else{
                tot-=curr;
            }
            prev=curr;
        }
        return tot;
    }
}